### ISSUE TITLE
Feature request: Auto discovery

### ISSUE THREAD

**Issue by LukeOwncloud** [owncloud/core#12318]
I am assuming that many ownCloud servers are accessible inside local networks.

If that is the case then easy service discovery is possible: http://en.wikipedia.org/wiki/Zero-configuration_networking#Service_discovery

So if a new client is installed on any device (smartphone or PC) and it is inside the local network, it can automatically detect that a local OC server is available. This server is then contacted and it can tell the client its official address (note: this address can be an external address) which is configured in the `config.php`. (Either the first external address from `trusted_domains` can be determined or probably better: A new entry is added, e.g., `public_url`.)

**Comment by PVince81** [comment]
A script that permanently binds is not doable with PHP. It would require an admin to setup a service to run said PHP script in parallel with the web server. If the admin anyway needs to setup services, they could rather simply create a zeroconf service file instead.

Closing as this is out of scope for OC / core.
